discord-bot/bot/settings.py

```diff
@@ -0,0 +1,15 @@
+# -*- coding: utf-8 -*-
+"""Configuration for the bot."""
+from pydantic import BaseSettings, Field
+
+
+class Settings(BaseSettings):
+    """Settings for the bot."""
+
+    token: str = Field(env="TOKEN", default="")
+    declare_global_commands: int = Field(env="DECLARE_GLOBAL_COMMANDS", default=0)
+    owner_ids: list[int] = Field(env="OWNER_IDS", default_factory=list)
+    prefix: str = Field(env="PREFIX", default="./")
+
+    class Config(BaseSettings.Config):
+        env_file = ".env"
```

Comment:
Could you please add `case_sensitive = False`, I guess it is default in pydantic but just to be explicit (since unix env names normally capitalized).

Reply:
1. I don't think there's a need to inherit form `BaseSettings.Config`, I think they're merged automatically
2. maybe it's worth (in the future) to make a base settings class of our own where we have things like case sensitivity and the presence of a `.env` file in.

Reply:
> 1. I don't think there's a need to inherit form `BaseSettings.Config`, I think they're merged automatically
> 2. maybe it's worth (in the future) to make a base settings class of our own where we have things like case sensitivity and the presence of a `.env` file in.

When I don't inherit, I get an error from my type-checker
```
"Config" overrides symbol of same name in class "BaseSettings"
  "discord-bot.bot.settings.Settings.Config" is incompatible with "pydantic.env_settings.BaseSettings.Config"
  Type "Type[Config]" cannot be assigned to type "Type[Config]"

Pylance[reportIncompatibleVariableOverride]
```
The message is a bit cryptic, but because we are subclassing `pydantic.BaseSettings`, we need to keep the types the same (including `class Config` which inherits from `pydantic.BaseConfig`, but we can't just inherit from `pydantic.BaseConfig`, we have to inherit from `pydantic.BaseSettings.Config`). This "defining a class inside a class" always confused me a bit, but I guess it works. I could add a `# type: ignore` comment, but inheriting make more sense to me.